var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -0,0 +1,288 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+import socket
+import os
+
+import llnl.util.tty as tty
+
+
+def cmake_cache_entry(name, value):
+    """
+    Helper that creates CMake cache entry strings used in
+    'host-config' files.
+    """
+    return 'set("%s" "%s" CACHE PATH "")\n\n' % (name, value)
+
+
+class Conduit(Package):
+    """Conduit is an open source project from Lawrence Livermore National
+    Laboratory that provides an intuitive model for describing hierarchical
+    scientific data in C++, C, Fortran, and Python. It is used for data
+    coupling between packages in-core, serialization, and I/O tasks."""
+
+    homepage = "http://software.llnl.gov/conduit"
+    url = "https://github.com/LLNL/conduit/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', 'd595573dedf55514c11d7391092fd760')
+
+    version('github-master', git='https://github.com/LLNL/conduit.git')
+
+    ###########################################################################
+    # package variants
+    ###########################################################################
+
+    variant("shared", default=True, description="Build Conduit as shared libs")
+
+    variant("cmake", default=True,
+            description="Build CMake (if off, attempt to use cmake from PATH)")
+
+    # variants for python support
+    variant("python", default=True, description="Build Conduit Python support")
+
+    # variants for i/o
+    variant("mpi", default=True, description="Build Conduit MPI Support")
+    variant("hdf5", default=True, description="Build Conduit HDF5 support")
+    variant("silo", default=True, description="Build Conduit Silo support")
```

Comment:
It's often better to turn off options by default, especially if they introduce a lot of dependencies.  But it's a judgement call; you should think about the options most "typical" installations might want.  If your application is different, you can set that up in `packages.yaml`.